Source/WebCore/platform/audio/gstreamer/AudioFileReaderGStreamer.cpp

```diff
@@ -182,14 +171,40 @@ GstFlowReturn AudioFileReader::handleSample(GstAppSink* sink)
         break;
     default:
         GST_WARNING("Unhandled channel: %d", GST_AUDIO_INFO_POSITION(&info, 0));
-        return GST_FLOW_ERROR;
+        return { };
     };
+    return channelId;
+}
+
+GstFlowReturn AudioFileReader::handleSample(GstAppSink* sink)
+{
+    auto sample = adoptGRef(gst_app_sink_try_pull_sample(sink, 0));
+    if (!sample)
+        return gst_app_sink_is_eos(sink) ? GST_FLOW_EOS : GST_FLOW_ERROR;
 
-    channelId++;
-    if (channelId == 1)
+    GstBuffer* buffer = gst_sample_get_buffer(sample.get());
+    if (!buffer)
+        return GST_FLOW_ERROR;
+
+    GstCaps* caps = gst_sample_get_caps(sample.get());
+    if (!caps)
+        return GST_FLOW_ERROR;
+
+    auto channelType = channelTypeFromCaps(caps);
+    if (!channelType)
+        return GST_FLOW_ERROR;
+
+    RELEASE_ASSERT(m_firstChannelType);
```

Comment:
I would make this a `GST_FLOW_ERROR` as well, maybe combined with `ASSERT_NOT_REACHED` if you want to be in the safer side or just an `ASSERT`